cpp/src/arrow/tensor.h

```diff
@@ -102,6 +102,8 @@ class ARROW_EXPORT Tensor {
 
   bool Equals(const Tensor& other) const;
 
+  size_t CountNonZero() const;
```

Comment:
Please add a docstring for this. Also, this should return `int64_t` like `size()` does.

Reply:
Done